pkg/api/v1alpha1/vspheredatacenterconfig.go

```diff
@@ -77,3 +77,36 @@ func validatePath(foldType folderType, folderPath string, datacenter string) err
 
 	return nil
 }
+
+func validateFailureDomains(vdc *VSphereDatacenterConfig) error {
+	failureDomains := vdc.Spec.FailureDomains
+	for _, fd := range failureDomains {
+		if len(fd.Name) <= 0 {
+			return fmt.Errorf("in the FailureDomain: %v, name is not set or is empty", fd)
```

Comment:
This reads better imo, unless you are getting this structure from something else.
```suggestion
			return fmt.Errorf("name is not set or is empty in FailureDomain %v", fd)
```